group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -9607,6 +9607,151 @@ public void testOnConsumerGroupStateTransitionOnLoading() {
         verify(context.metrics, times(1)).onConsumerGroupStateTransition(ConsumerGroup.ConsumerGroupState.EMPTY, null);
     }
 
+    @Test
+    public void testMaybeUpgradeEmptyGroup() {
```

Comment:
`testMaybeUpgradeEmptyGroup` and `testMaybeDowngradeEmptyGroup` seem a bit redundant now that we have the others. Do they still bring something?